html/beps/bep_encrypted_data.rst

```diff
@@ -0,0 +1,297 @@
+In other cases the content may be meant for open distribution within a community without intent of announcing the content to the whole world. This is analogous to web content that is open to human visitors but requests via robots.txt[robots]_ that it should not be announced to the world by web crawlers.   
+
+
+While the private flag [BEP-27]_ may be sufficient in a controlled environment to prevent information about the torrent (e.g. its infohash) from escaping and thus preventing others from connecting to the swarm this is a very brittle form of security which also prevents the use of public infrastructure such as open trackers, PEX or the DHT.
+Similarly Message Stream Encryption provides limited protection from passive eavesdroppers on the network layer but does not prevent the infohash from escaping.   
+
+
+Instead of attempting to restrict access to the swarm or metadata this BEP proposes to make all data opaque to 3rd parties by encrypting it with a shared secret that is not available through any torrent-related protocol, i.e. must be obtained separately by the user.
+
+In principle the same properties can be provided by simply storing the data in an encrypted archive and using a nondescript filename, but that requires users to store the data twice or to use additional filesystem layers to transparently access the data, which is even more cumbersome when encryption is involved. It also prevents bittorrent clients from reusing already-downloaded files in a multi-file torrent. 
+
+Metadata format
+===============
+
+
+.. parsed-literal::
+
+    {
+      info: {
+        bepXX: {
+          mac: *<32bytes of hmac output (string)>*,
+          salt: *<32bytes of random binary data (string)>*,
+          shadow: *<optional, bencoded-then-encrypted dictionary (string)>*
+          v: *<version (integer)>*,
+        },
+        length *or* files: *<unchanged>*,
+        name: *<public name (string)>*,
+        piece length: *<unchanged>*
+        pieces: *<N*20 bytes, piece hashes of the payload ciphertext (string)>*
+      },
+    }
+
+
+``salt``
+  the random data must be generated by a cryptographically secure RNG to avoid IV reuse.
+
+``v``
+  The protocol version used to encrypt the torrent, currently *1*. New versions may be introduced by updates to this BEP if cryptographic weaknesses necessitate incompatible changes. Implementations should check if they support the version indicated in the metadata file and otherwise inform the user that they can download the data but not decrypt it.
+
+``shadow``
+  bencoded-then-encrypted dictionary of key-value pairs that shadow entries in the info dictionary.
+  If it is absent only the payload is encrypted and all info dictionary entries are non-shadowed.
+  Implementations should only shadow a whitelist of keys which they have a shadowing strategy and ignore other keys.
+  Shadowable keys suggested by this BEP: ``length, files, name, comment``.
```

Comment:
it might look nicer to just make the names literals, not the commas separating them